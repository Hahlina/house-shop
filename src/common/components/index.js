export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
