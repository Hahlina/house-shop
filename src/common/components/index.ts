export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
