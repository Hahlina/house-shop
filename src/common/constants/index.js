export * from './config';
export * from './apiRoutes';
export * from './initValues';
export * from './appRoutes';
