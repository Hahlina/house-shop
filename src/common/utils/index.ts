export * from './convertPrice';
export * from './capitalizeFirstLetter';
