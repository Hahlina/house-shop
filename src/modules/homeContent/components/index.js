export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
