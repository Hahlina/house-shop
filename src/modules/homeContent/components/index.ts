export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
