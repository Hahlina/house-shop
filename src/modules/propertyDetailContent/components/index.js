export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
