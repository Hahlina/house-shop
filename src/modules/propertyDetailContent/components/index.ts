export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
