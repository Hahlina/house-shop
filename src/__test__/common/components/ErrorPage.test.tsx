import { ErrorPage } from 'common/components';

import { renderWithRoutes, screen } from 'common/utils/test';
import { APP_ROUTES } from 'common/constants';

describe('ErrorPage component', () => {
    it('should renders the ErrorPage with the 404 image', () => {
        renderWithRoutes(<ErrorPage />);
        const image = screen.getByRole('img');
        expect(image).toBeInTheDocument();
    });

    it('should renders the ErrorPage with the correct error code', () => {
        renderWithRoutes(<ErrorPage />);
        const errorCode = screen.getByText(/404/i);
        expect(errorCode).toBeInTheDocument();
    });

    it('should renders the ErrorPage with the "Go home" button', () => {
        renderWithRoutes(<ErrorPage />);
        const goHomeButton = screen.getByRole('button', { name: /go home/i });
        expect(goHomeButton).toBeInTheDocument();
    });

    it('should renders the ErrorPage with a link to the home page', () => {
        renderWithRoutes(<ErrorPage />);
        const goHomeButton = screen.getByRole('button', { name: /go home/i });
        expect(goHomeButton.closest('a')).toHaveAttribute('href', APP_ROUTES.HOME);
    });

    it('should take a snapshot', () => {
        const { container } = renderWithRoutes(<ErrorPage />);
        expect(container).toMatchSnapshot();
    });
});
