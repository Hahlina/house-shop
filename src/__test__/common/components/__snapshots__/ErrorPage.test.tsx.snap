// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage component should take a snapshot 1`] = `
<div>
  <div
    class="sc-eDLKkx mMCzY"
  >
    <div
      class="sc-jTQCzO bRQGNi"
    >
      <img
        src="/images/404.png"
        style="visibility: visible; transition: visibility 200ms ease-out 700ms;"
      />
      <div
        class="sc-gLLuof fuTWTb"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 code css-ahj2mt-MuiTypography-root"
        >
          Opps...404
        </p>
        <a
          href="/"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-1sszw6i-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Go home
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
